jcache/remote/src/main/java/org/infinispan/jcache/remote/RemoteCacheWithStats.java

```diff
@@ -0,0 +1,88 @@
+package org.infinispan.jcache.remote;
+
+import java.util.concurrent.TimeUnit;
+
+import org.infinispan.client.hotrod.RemoteCache;
+
+public class RemoteCacheWithStats<K, V> extends RemoteCacheWrapper<K, V> {
+   private LocalStatistics stats;
+
+   public RemoteCacheWithStats(RemoteCache<K, V> delegate, LocalStatistics stats) {
+      super(delegate);
+      this.stats = stats;
+   }
+
+   @Override
+   public V get(Object key) {
+      V v = delegate.get(key);
+      if (v == null) {
+         stats.incrementCacheMisses();
+      } else {
+         stats.incrementCacheHits();
+      }
+      stats.incrementCacheGets();
+      return v;
+   }
+
+   @Override
+   public V put(K key, V value) {
+      V v = delegate.put(key, value);
+      stats.incrementCachePuts();
+      return v;
+   }
+
+   @Override
+   public V put(K key, V value, long lifespan, TimeUnit unit) {
+      V v = delegate.put(key, value, lifespan, unit);
+      stats.incrementCachePuts();
+      return v;
+   }
+
+   @Override
+   public V putIfAbsent(K key, V value) {
+      V result = delegate.putIfAbsent(key, value);
+      if (result == null) {
+         stats.incrementCachePuts();
+      }
+      return result;
+   }
+
+   @Override
+   public V replace(K key, V value) {
+      V v = delegate.replace(key, value);
+      if (v != null) {
+         stats.incrementCacheHits();
```

Comment:
Right, it's doing that for JCache only methods: ```getAndRemove```, ```getAndReplace```. Exception to the rule is ```replace``` which is uses ```cacheWithoutStats``` wrapper 1st....